File: Programming101/week0/is_number_balanced.py
def is_number_balanced(n):
    n = str(n)
    left_sum = 0
    right_sum = 0
    for dig in range(0, (len(n) // 2)):
        left_sum += int(n[dig])
        right_sum += int(n[-dig-1])
    return left_sum == right_sum


def main():
    print(is_number_balanced(9))
    print(is_number_balanced(11))
    print(is_number_balanced(13))
    print(is_number_balanced(121))
    print(is_number_balanced(4518))
    print(is_number_balanced(28471))
    print(is_number_balanced(1238033))


if __name__ == '__main__':
    main()

File: FrontendJavaScript/week0/2-A-Set-Of-Problems/set1/isNumberBalanced.js
'use strict';

var isNumberBalanced = function(n) {
  n = n.toString();
  var left_sum = 0;
  var right_sum = 0;
  for (var i = 0; i < n.length - 1 - i; i++) {
    left_sum += parseInt(n[i]);
    right_sum += parseInt(n[n.length - 1- i]);
  }
  if (left_sum === right_sum) {
    return true;
  }
  return false;
};

console.log(isNumberBalanced(9) === true);
console.log(isNumberBalanced(11) === true);
console.log(isNumberBalanced(13) === false);
console.log(isNumberBalanced(121) === true);
console.log(isNumberBalanced(4518) === true);
console.log(isNumberBalanced(28471) === false);
console.log(isNumberBalanced(1238033) === true);
